backend/test/modules/doubles-tournament/core/application/BookTableAndPublishEventAfterCallingMatch.spec.ts

```diff
@@ -0,0 +1,187 @@
+        return CommandResult.success();
+      },
+    };
+
+    const commandBus = new InMemoryCommandBus();
+    commandBus.registerHandler(BookTournamentTable, bookTournamentTableCommandHandler);
+    const executeSpy = jest.spyOn(commandBus, 'execute');
+
+    const doublesTournament = testDoublesTournamentsModule(currentTime, entityIdGen, commandBus);
+
+    //When
+    const callMatch = new CallMatch({
+      tournamentId: tournamentId,
+      matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+      tableNumber: tableNumber,
+    });
+    const commandResult = await doublesTournament.executeCommand(callMatch);
+
+    //Then
+    expect(commandResult.isSuccess()).toBeTruthy();
+    expect(doublesTournament.lastPublishedEvent()).toStrictEqual(
+      new MatchWasCalled({
+        occurredAt: currentTime,
+        tournamentId: tournamentId,
+        matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+        tableNumber: tableNumber,
+      }),
+    );
+
+    expect(executeSpy).toHaveBeenCalledTimes(2);
+
+    const bookFromAvailableTables = new BookTournamentTable(tournamentId, tableNumber);
+    expect(commandBus.execute).toHaveBeenLastCalledWith(bookFromAvailableTables);
+  });
+
+  it('Then publish event and execute BookTournamentTable command from different module | table has been already booked', async () => {
+    //Given
+    const bookTournamentTableCommandHandler: CommandHandler<BookTournamentTable> = {
+      async execute(command: BookTournamentTable): Promise<CommandResult> {
+        return CommandResult.success();
+      },
+    };
+
+    const commandBus = new InMemoryCommandBus();
+    commandBus.registerHandler(BookTournamentTable, bookTournamentTableCommandHandler);
+    const executeSpy = jest.spyOn(commandBus, 'execute');
+
+    const doublesTournament = testDoublesTournamentsModule(currentTime, entityIdGen, commandBus);
+
+    //When
+    const callMatch = new CallMatch({
+      tournamentId: tournamentId,
+      matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+      tableNumber: tableNumber,
+    });
+    const commandResult = await doublesTournament.executeCommand(callMatch);
+
+    //Then
+    expect(commandResult.isSuccess()).toBeTruthy();
+    expect(doublesTournament.lastPublishedEvent()).toStrictEqual(
+      new MatchWasCalled({
+        occurredAt: currentTime,
+        tournamentId: tournamentId,
+        matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+        tableNumber: tableNumber,
+      }),
+    );
+
+    expect(executeSpy).toHaveBeenCalledTimes(2);
+
+    const bookFromAvailableTables = new BookTournamentTable(tournamentId, tableNumber);
+    expect(commandBus.execute).toHaveBeenLastCalledWith(bookFromAvailableTables);
+  });
+
+  //--------------------------------------------------------------------------------------------------------------------
```

Comment:
@nowakprojects and everybody - pytanie!
Wg mnie to poniżej kreski jest do usunięcia (są to trochę rozszerzone testy do wersji testów powyżej kreski, ale "za bardzo rozszerzone"). Nie potrzebujemy tutaj testować, czy BookTournamentTable nam się uda, czy nie, tylko, czy został wywołany na nim execute, zgadza się? Jeśli dobrze myślę to te dwa TC poniżej kreski mogę usunąć, Si?

Reply:
Mhm, nie testujesz czy sie uda czy nie, ale powinienes testowac jak reaguje Twój commandhandler na to, ze BookTournamentTable sie udal albo nie udał.